lib/reboot_cmds.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+"""
+Some uaclient operations cannot be fully completed by running a single
+command. For example, when upgrading uaclient from trusty to xenial,
+we may have a livepatch change in the contract, allowing livepatch to be
+enabled on xenial. However, during the upgrade we cannot install livepatch on
+the system, only after a reboot.
+
+To allow uaclient to postpone commands that need to be executed in a system boot,
+we are using this script, which will basically try to reprocess contract deltas
+on boot time.
+"""
+import logging
+import os
+import sys
+
+from uaclient import config
+
+from uaclient.util import subp, ProcessExecutionError
+from uaclient.cli import setup_logging, assert_lock_file
+
+
+BASE_PATH = "/var/lib/ubuntu-advantage/"
+REBOOT_CMD_MARKER_FILE = BASE_PATH + "marker-reboot-cmds-required"
+
+
+def run_command(cmd):
+    try:
+        out, _ = subp(cmd.split())
+        logging.debug("Successfully executed cmd: {}".format(cmd))
+    except ProcessExecutionError as exec_error:
+        msg = (
+            "Failed running cmd: {}\n"
+            "Return code: {}\n"
+            "Stderr: {}\n"
+            "Stdout: {}".format(
+                cmd, exec_error.exit_code, exec_error.stderr, exec_error.stdout
+            )
+        )
+        logging.debug(msg)
+        sys.exit(1)
+
+
+def refresh_contract(args, cfg):
+    cmd = "ua refresh"
+    run_command(cmd=cmd)
+
+
+@assert_lock_file("ua-reboot-process-deltas")
+def process_remaining_deltas(args, cfg):
+    cmd = "/usr/bin/python3 /usr/lib/ubuntu-advantage/upgrade_lts_contract.py"
+    run_command(cmd=cmd)
+
+
+def main(args, cfg):
+    setup_logging(logging.INFO, logging.DEBUG)
+
+    if not cfg.is_attached:
+        logging.debug("Skiping reboot_cmds. Machine is unattached")
+
+        if os.path.exists(REBOOT_CMD_MARKER_FILE):
+            os.remove(REBOOT_CMD_MARKER_FILE)
```

Comment:
Let's use our del_file utility func to avoid this duplication of check/remove
```suggestion
        util.del_file(REBOOT_CMD_MARKER_FILE)
```